RustFxA/FirefoxAccountSignInViewController.swift

```diff
@@ -283,14 +283,14 @@ class FirefoxAccountSignInViewController: UIViewController, Themeable {
 // MARK: QRCodeViewControllerDelegate Functions
 extension FirefoxAccountSignInViewController: QRCodeViewControllerDelegate {
     func didScanQRCodeWithURL(_ url: URL) {
-        let askForPermission = OnboardingNotificationCardHelper().askForPermissionDuringSync(
-            isOnboarding: telemetryObject == .onboarding)
+        let isOnboarding = telemetryObject == .onboarding
+        let shouldAskForPermission = !OnboardingNotificationCardHelper().notificationCardIsInOnboarding() || !isOnboarding
```

Comment:
Looks like this code is a duplicate of the above. Let's put it in a method and call it from both places.